### PR TITLE
[stable] fix: ensure tutorials support settings.dict_aux_operators

### PR DESCRIPTION
**Pull request by mrossinek**
### Summary

This PR ensures that the stable tutorials support `settings.dict_aux_operators = True`

### CODE REVIEW

**Comment by woodsp-ibm** [comment]
So even though there is a deprecation warning for using indexed based access it does not come out in these tutorials. I'm not sure I am entirely happy with exposing the access done either way in these, especially since we are saying we are switching away from list based to dict. Maybe we should set the settings in the tutorial and just do the dict based logic there?

**Comment by mrossinek** [comment]
I'm fine with any approach that will make things consistent. Enabling dict-based operators via the setting in all tutorials seems like a good approach, too :+1: 
I can update this PR accordingly later today.

File: docs/tutorials/01_electronic_structure.ipynb [hunk 1]
```diff
@@ -190,7 +201,7 @@
    "source": [
     "es_problem = ElectronicStructureProblem(driver)\n",
     "second_q_op = es_problem.second_q_ops()\n",
-    "print(second_q_op[0])"
+    "print(second_q_op[\"ElectronicEnergy\"])"
```
Comment (woodsp-ibm) [inline docs/tutorials/01_electronic_structure.ipynb:204]: In theory we have the `main_property_name` on the problem that avoids knowing/hard-coding a string value explicitly here. Should we be using that, or at least showing that here. I know the main goal was to go to the dictionary, but we had that way to allow users to get the main property without having to know. Here for instance we could get that name, print it as well as printing the operator.

ParticleNumber usage below evidently cannot be altered.

Comment (mrossinek) [reply docs/tutorials/01_electronic_structure.ipynb:204]: I figured that this is really just a temporary fix until the 0.5 release. At that point `main_property_name` won't exist any more. So maybe we can keep it simple like here?

**Review by woodsp-ibm — APPROVED** [review]

**Pull request merged**